src/controllers/user.controller.js

```diff
@@ -0,0 +1,69 @@
+import bcrypt from 'bcrypt';
+
+import models from '../db/models';
+import sender from '../services/email';
+import Response from '../utils/response.utils';
+import UserUtils from '../utils/user.utils';
+import JWTService from '../services/jwt.service';
+
+const { User } = models;
+
+/**
+ * This class creates the user controller
+ */
+export default class UserController {
+/**
+ * @param {object} req The user's signup details
+ * @param {object} res The user's details returned after signup
+ * @returns {object} A signed up user
+ */
+  static async signup(req, res) {
+    try {
+      const hash = await bcrypt.hash(req.body.password, 10);
+      const userData = UserUtils.getUserSignupData(req.body);
+      const user = await User.create({
+        ...userData,
+        password: hash
+      }, { returning: true });
+
+      // parameter(s) to be passed to the sendgrid email template
+      const payload = { user };
+      await sender.sendEmail(process.env.SENDER_EMAIL, user.email, 'signup_template', payload);
+
+      const userToken = JWTService.generateToken(user);
+      return Response.Success(res, {
+        id: user.id,
+        ...userData,
+        role: user.role,
+        is_verified: user.is_verified,
+        token: userToken
+      }, 201);
+    } catch (error) {
+      return Response.InternalServerError(res, 'Could not signup user');
+    }
+  }
+
+  /**
+   * Handles signin requests
+   * @param {ServerRequest} req
+   * @param {ServerResponse} res
+   * @returns {ServerResponse} response
+   */
+  static signin(req, res) {
+    const signinError = { message: 'Incorrect email or password' };
+    UserUtils.findUserByEmail(req.body.email)
```

Comment:
Do you think this could have been less efficient if you had used the Sequelize's `User.findOne`? or `User.findAll({limit: 1})` ?

Reply:
This has been taken care of